huntsman/observatory.py

```diff
@@ -232,202 +233,61 @@ def autofocus_cameras(self, *args, **kwargs):
 
         return result
 
-    def take_evening_flats(self,
-                           alt=None,
-                           az=None,
-                           min_counts=5000,
-                           max_counts=15000,
-                           bias=1000,
-                           max_exptime=60.,
-                           camera_list=None,
-                           target_adu_percentage=0.5,
-                           max_num_exposures=10,
-                           *args, **kwargs
-                           ):
-        """Take flat fields
+    def take_flat_fields(self, camera_names=None, **kwargs):
+        """
+        Take flat fields for each camera in each filter, respecting filter priority.
 
         Args:
-            alt (float, optional): Altitude for flats
-            az (float, optional): Azimuth for flats
-            min_counts (int, optional): Minimum ADU count
-            max_counts (int, optional): Maximum ADU count
-            bias (int, optional): Default bias for the cameras
-            max_exptime (float, optional): Maximum exposure time before stopping
-            camera_list (list, optional): List of cameras to use for flat-fielding
+            camera_names (list, optional): List of camera names to take flats with.
+            alt (float, optional): Altitude for flats.
+            az (float, optional): Azimuth for flats.
+            min_counts (int, optional): Minimum ADU count.
+            max_counts (int, optional): Maximum ADU count.
+            bias (int, optional): Default bias for the cameras.
+            max_exptime (float, optional): Maximum exposure time before stopping.
+            camera_list (list, optional): List of cameras to use for flat-fielding.
             target_adu_percentage (float, optional): Exposure time will be adjust so
-                that counts are close to: target * (`min_counts` + `max_counts`). Default
-                to 0.5
-            max_num_exposures (int, optional): Maximum number of flats to take
-            *args (TYPE): Description
-            **kwargs (TYPE): Description
-
+                that counts are close to: target * (`min_counts` + `max_counts`). Defaults
+                to 0.5.
+            max_num_exposures (int, optional): Maximum number of flats to take.
         """
-        target_adu = target_adu_percentage * (min_counts + max_counts)
-
-        image_dir = self.config['directories']['images']
-
-        flat_obs = self._create_flat_field_observation(alt=alt, az=az)
-        exptimes = {cam_name: [1. * u.second] for cam_name in camera_list}
-
-        # Loop until conditions are met for flat-fielding
+        if camera_names is None:
+            camera_names = list(self.cameras.keys())
+        cameras = [self.cameras[cam_name] for cam_name in camera_names]
+
+        # Create priority-ordered list of filter names for each camera
+        filter_names = {}
+        for cam_name, camera in zip(camera_names, cameras):
```

Comment:
Neater to work with camera dictionaries here, I think.

```suggestion
        if camera_names is None:
            cameras = self.cameras
        else:
            cameras = {cam_name:self.cameras[cam_name] for cam_name in camera_names}

        # Create priority-ordered list of filter names for each camera
        filter_names = {}
        for cam_name, camera in cameras.items():
```